### PR TITLE
This Engineer Disabled DevRegistry - You Won't Believe What Happened Next!

### PR DESCRIPTION
**Pull request by rozenmd**
Closes https://github.com/cloudflare/wrangler2/issues/2353

### CODE REVIEW

**Review by JacobMGEvans — APPROVED** [review]

File: packages/wrangler/src/api/dev.ts [hunk 1]
```diff
@@ -80,8 +81,11 @@ export async function unstable_dev(
 	options?: DevOptions,
 	apiOptions?: DevApiOptions
 ): Promise<UnstableDevWorker> {
-	const { testMode = true, disableExperimentalWarning = false } =
-		apiOptions || {};
+	const {
+		testMode = true,
+		disableExperimentalWarning = false,
+		experimentalDisableDevRegistry = false,
+	} = apiOptions || {};
```
Comment (penalosa) [inline packages/wrangler/src/api/dev.ts:88]:
```suggestion
	} = apiOptions ?? {};
```

**Review by penalosa — CHANGES_REQUESTED** [review]
I wonder if this might be a good opportunity to destructure things less? Most of the changeset of this PR comes from destructuring `apiOptions`, and I wonder if keeping it as `apiOptions` throughout would be cleaner, and allow for simpler adding of new options

File: packages/wrangler/package.json [hunk 2]
```diff
@@ -63,9 +63,9 @@
 		"emit-types": "tsc -p tsconfig.emit.json && node -r esbuild-register scripts/emit-types.ts",
 		"prepublishOnly": "SOURCEMAPS=false npm run build",
 		"start": "npm run bundle && cross-env NODE_OPTIONS=--enable-source-maps ./bin/wrangler.js",
-		"test": "npm run assert-git-version && node --expose-gc ../../node_modules/.bin/jest --runInBand --logHeapUsage --silent=false --verbose=true",
+		"test": "npm run assert-git-version && cross-env NODE_OPTIONS=--max_old_space_size=8192 jest --silent=false --verbose=true",
```
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wrangler/package.json:66]: Boosting the memory in tests for then removing `--runInBand` which causes the tests to run concurrently. So the change would give Jest more memory to run tests in parallel.

File: packages/wrangler/src/__tests__/api-devregistry.test.js [hunk 3]
```diff
@@ -1,26 +1,36 @@
 import { unstable_dev } from "../api";
 import { fetch } from "undici";
 
-// jest.unmock("undici");
+jest.unmock("undici");
```
[the remark this answers is not in the document]
Comment (rozenmd) [reply packages/wrangler/src/__tests__/api-devregistry.test.js:4]: This test shouldn't use any mocking - I want an actual HTTP server running, and I want to actually fetch from it

File: fixtures/local-mode-tests/tests/headers.test.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ describe.skip("worker", () => {
 		worker = await unstable_dev(
 			path.resolve(__dirname, "..", "src", "headers.ts"),
 			{},
-			{ disableExperimentalWarning: true }
+			{ disableExperimentalWarning: true, experimentalDisableDevRegistry: true }
```
[the remark this answers is not in the document]
Comment (rozenmd) [reply fixtures/local-mode-tests/tests/headers.test.ts:22]: Yes - DevRegistry should be enabled in `unstable_dev` when testing multi-worker configurations, such as https://developers.cloudflare.com/workers/wrangler/api/#multi-worker-example

**Comment by penalosa** [comment]
@rozenmd Passing it down through the call stack as `apiOptions`, and using it as `apiOptions. experimentalDisableDevRegistry` when an option is needed.

**Comment by rozenmd** [comment]
@penalosa can you share an example of what you mean?

To me that sounds like it would save like ~3 LOC in https://github.com/cloudflare/wrangler2/pull/2342/files#diff-3cfa6108cd2ddf7c511b1d3dc125143eb1a56aea1f1807bfc9a8e8c6aba5f45fR84-R88 - most of the changes in the PR are tests

**Comment by penalosa** [comment]
Just don't want to end up in the same situation of the regular dev with loads of options being re-passed down in every function. Not a big deal here though, I was more concerned with the `unmock` of `undici`, but it sounds like that's okay. Approving pending Windows tests passing

**Review by penalosa — APPROVED** [review]

**Comment by rozenmd** [comment]
Closing this PR in favour of another that uses `disableDevRegistry` inside the new `experimental` object

**Pull request closed without merging**